go/roothash/api/api.go

```diff
@@ -169,42 +170,14 @@ func SanityCheckBlocks(blocks map[signature.PublicKey]*block.Block) error {
 	return nil
 }
 
-// checkBlocksForGenesis examines the blocks for extra properties specific to genesis state.
-func checkBlocksForGenesis(blocks map[signature.PublicKey]*block.Block) error {
-	for _, blk := range blocks {
-		hdr := blk.Header
-
-		if hdr.HeaderType != block.Normal {
-			return fmt.Errorf("roothash: sanity check failed: invalid block header type %v", hdr.HeaderType)
-		}
-
-		if !hdr.PreviousHash.IsEmpty() {
-			return fmt.Errorf("roothash: sanity check failed: non-empty previous hash")
-		}
-
-		if len(hdr.StorageSignatures) != 0 {
-			return fmt.Errorf("roothash: sanity check failed: non-empty storage signatures")
-		}
-
-		if len(hdr.Messages) != 0 {
-			return fmt.Errorf("roothash: sanity check failed: non-empty roothash messages")
-		}
-	}
-	return nil
-}
-
 // SanityCheck does basic sanity checking on the genesis state.
-func (g *Genesis) SanityCheck() error {
+// isGenesis is true, if it is called during the Tendermint's InitChain.
+func (g *Genesis) SanityCheck(isGenesis bool) error {
```

Comment:
When would `SanityCheck` on a genesis document be called with `isGenesis` set to false? I think this argument should be removed and the below `SanityCheck` always called with `true`.

Reply:
Ok, I left the internal RuntimeGenesis `isGenesis` parameter alone and only remove it from the roothash's one.